charts/dragonfly/templates/manager/manager-configmap.yaml

```diff
@@ -40,18 +40,20 @@ data:
         migrate: {{ .Values.externalMysql.migrate }}
         {{- end }}
       redis:
-        {{- if and .Values.redis.enable (empty .Values.externalRedis.host) }}
+        {{- if .Values.redis.enable }}
         addrs:
         - {{ .Release.Name }}-{{ default "redis" .Values.redis.fullname }}-master.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}:{{ .Values.redis.master.service.ports.redis }}
         host: {{ .Release.Name }}-{{ default "redis" .Values.redis.fullname }}-master.{{ .Release.Namespace }}.svc.{{ .Values.clusterDomain }}
         port: {{ .Values.redis.master.service.ports.redis }}
         password: {{ .Values.redis.auth.password }}
         {{- else }}
-        addrs:
-        - {{ .Values.externalRedis.host }}:{{ .Values.externalRedis.port }}
-        host: {{ .Values.externalRedis.host }}
-        port: {{ .Values.externalRedis.port }}
+        addrs: {{ .Values.externalRedis.addrs }}
```

Comment:
addrs is an array, you should use 'toYaml'
{{ toYaml .Values.externalRedis.Addrs | indent 10 }}

Reply:
{{ toYaml .Values.externalRedis.addrs | indent 10 }}